### PR TITLE
added helm formulae

### PR DESCRIPTION
**Pull request by skvijay007**
added helm formulae from 3.30 to 3.5.2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,52 @@
...
+
+
+## Update with new formula
+
+When new version of Helm or Kubernetes-cli is released, brew will update its packages. To view all the previous versions of the tool, we can visit the brew histories of [Helm](https://github.com/Homebrew/homebrew-core/commits/7877b4bbcfa48028349a5cbed0d2f6ef5dbcc84b/Formula/helm.rb) and [Kubernetes-cli](https://github.com/Homebrew/homebrew-core/commits/674e098c3efce9196d91397d325c66d488aacc6b/Formula/kubernetes-cli.rb) to view list of all the previous versions.
+
+
+### Updating tap
+
+1. Create a file `helm@<x.x.x>.rb` or `Kubernetes-cli@<x.x.x>`
```
Comment (GavinOsborn) [inline README.md]: Why is 1 an `.rb` file and the other have no extension?
Is this significant? Is it a typo?

File: README.md [hunk 2]
```diff
@@ -1 +1,52 @@
...
+
+    3. Copy the content of the formula from the bottle file to the newly created tap file.
+
+    ![Bottle file](./assets/bottlefile.png)
+
+3. Change the formula class name
+
+Class name of the formula should be changed to support installing specific version of the formula.
+
+***File name should be helm@<x.x.x>.rb or Kubernetes-cli@<x.x.x>
```
Comment (GavinOsborn) [inline README.md]: Is this `AT<version` convention something that we have created? Or a convention documented by homebrew?

File: README.md [hunk 3]
```diff
@@ -1 +1,52 @@
...
+
+
+`` e.g for helm@3.10.1.rb file class name should be HelmAT3101,
+    for kubernetes-cli@1.18.8 file class name should be KubernetesCliAT1188 ``
+
+
+
+# How do I install these formulae?
+
+`brew install nintexplatform/homebrew-versions/<formula>`
```
Comment (GavinOsborn) [inline README.md:56]: Could you also give a complete example with a value for `<formula>` please :)

**Comment by GavinOsborn** [comment]
If I were doing the next version bump, how would I test it locally before merging?

**Comment by GavinOsborn** [comment]
I assume you added like 8 different Helm versions because you were testing?
Do we need to keep them all though?

File: README.md (hunk 1 above)
Comment (skvijay007) [reply README.md]: it's typo, will fix it

File: README.md (hunk 2 above)
Comment (skvijay007) [reply README.md]: That's the naming class naming convention need to be followed. Example existing formula in homebrew https://github.com/Homebrew/homebrew-core/blob/master/Formula/ansible%402.8.rb

**Comment by skvijay007** [comment]
> I assume you added like 8 different Helm versions because you were testing?
> Do we need to keep them all though?

As discussed earlier - option B, we are going to keep all the versions of the tool and choose which version to be installed in the script.

**Comment by skvijay007** [comment]
> If I were doing the next version bump, how would I test it locally before merging?

We can test locally by running brew command `brew install --build-from-source ./helm@3.5.2.rb`. I will update this step in readme.

**Comment by skvijay007** [comment]
> I assume you added like 8 different Helm versions because you were testing?
> Do we need to keep them all though?

**Review by GavinOsborn — COMMENTED** [review]
> > If I were doing the next version bump, how would I test it locally before merging?
> 
> We can test locally by running brew command `brew install --build-from-source ./helm@3.5.2.rb`. I will update this step in readme.

Sounds like something we should add to the docs :)

File: README.md (hunk 2 above)
Comment (GavinOsborn) [reply README.md]: Is there a link to some homebrew documentation that explains why?

**Comment by GavinOsborn** [comment]
> > I assume you added like 8 different Helm versions because you were testing?
> > Do we need to keep them all though?
> 
> As discussed earlier - option B, we are going to keep all the versions of the tool and choose which version to be installed in the script.

We discussed have multiple versions, yes - but does that mean we have to have _every_ version?
Or just the versions that we choose to adopt?

File: README.md (hunk 2 above)
Comment (skvijay007) [reply README.md]: https://docs.brew.sh/Versions.html

Comment (GavinOsborn) [reply README.md]: Cool. A useful link to include in the docs :)

**Comment by skvijay007** [comment]
> > > I assume you added like 8 different Helm versions because you were testing?
> > > Do we need to keep them all though?
> > 
> > 
> > As discussed earlier - option B, we are going to keep all the versions of the tool and choose which version to be installed in the script.
> 
> We discussed have multiple versions, yes - but does that mean we have to have _every_ version?
> Or just the versions that we choose to adopt?

We don't really need to have all versions as we will using only a specific version at a time. When we need to update the version in the script, we can eventually add that formula here and use it.

**Comment by petedv** [comment]
It looks like we will have a flat directory structure. Is that mandatory? 
Is it possible to have a directory per formula?

**Comment by skvijay007** [comment]
> It looks like we will have a flat directory structure. Is that mandatory?
> Is it possible to have a directory per formula?

Let me check if having multiple subfolders in Formula works.

**Comment by petedv** [comment]
> It looks like we will have a flat directory structure. Is that mandatory?
> Is it possible to have a directory per formula?

Nevermind i found an answer [here](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap)

>Tap formulae follow the same format as the core’s ones, and can be added under either the Formula subdirectory, the HomebrewFormula subdirectory or the repository’s root. The first available directory is used, other locations will be ignored. We recommend use of subdirectories because it makes the repository organisation easier to grasp, and top-level files are not mixed with formulae.

Do we want to use a Formula directory?

**Comment by skvijay007** [comment]
> > It looks like we will have a flat directory structure. Is that mandatory?
> > Is it possible to have a directory per formula?
> 
> Nevermind i found an answer [here](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap)
> 
> > Tap formulae follow the same format as the core’s ones, and can be added under either the Formula subdirectory, the HomebrewFormula subdirectory or the repository’s root. The first available directory is used, other locations will be ignored. We recommend use of subdirectories because it makes the repository organisation easier to grasp, and top-level files are not mixed with formulae.
> 
> Do we want to use a Formula directory?

It's good to have the Formula folder to avoid mixing up with other top-level files

File: README.md (hunk 1 above)
Comment (skvijay007) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (skvijay007) [reply README.md]: Added to readme

**Comment by skvijay007** [comment]
> > > > I assume you added like 8 different Helm versions because you were testing?
> > > > Do we need to keep them all though?
> > > 
> > > 
> > > As discussed earlier - option B, we are going to keep all the versions of the tool and choose which version to be installed in the script.
> > 
> > 
> > We discussed have multiple versions, yes - but does that mean we have to have _every_ version?
> > Or just the versions that we choose to adopt?
> 
> We don't really need to have all versions as we will using only a specific version at a time. When we need to update the version in the script, we can eventually add that formula here and use it.

Keeping only required version of helm and kubectl in tap

**Comment by GavinOsborn** [comment]
> > > It looks like we will have a flat directory structure. Is that mandatory?
> > > Is it possible to have a directory per formula?
> > 
> > 
> > Nevermind i found an answer [here](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap)
> > > Tap formulae follow the same format as the core’s ones, and can be added under either the Formula subdirectory, the HomebrewFormula subdirectory or the repository’s root. The first available directory is used, other locations will be ignored. We recommend use of subdirectories because it makes the repository organisation easier to grasp, and top-level files are not mixed with formulae.
> > 
> > 
> > Do we want to use a Formula directory?
> 
> It's good to have the Formula folder to avoid mixing up with other top-level files

Hey @skvijay007 did we action this?

**Comment by skvijay007** [comment]
> > > > It looks like we will have a flat directory structure. Is that mandatory?
> > > > Is it possible to have a directory per formula?
> > > 
> > > 
> > > Nevermind i found an answer [here](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap)
> > > > Tap formulae follow the same format as the core’s ones, and can be added under either the Formula subdirectory, the HomebrewFormula subdirectory or the repository’s root. The first available directory is used, other locations will be ignored. We recommend use of subdirectories because it makes the repository organisation easier to grasp, and top-level files are not mixed with formulae.
> > > 
> > > 
> > > Do we want to use a Formula directory?
> > 
> > 
> > It's good to have the Formula folder to avoid mixing up with other top-level files
> 
> Hey @skvijay007 did we action this?

**Comment by skvijay007** [comment]
> > > > It looks like we will have a flat directory structure. Is that mandatory?
> > > > Is it possible to have a directory per formula?
> > > 
> > > 
> > > Nevermind i found an answer [here](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap)
> > > > Tap formulae follow the same format as the core’s ones, and can be added under either the Formula subdirectory, the HomebrewFormula subdirectory or the repository’s root. The first available directory is used, other locations will be ignored. We recommend use of subdirectories because it makes the repository organisation easier to grasp, and top-level files are not mixed with formulae.
> > > 
> > > 
> > > Do we want to use a Formula directory?
> > 
> > 
> > It's good to have the Formula folder to avoid mixing up with other top-level files
> 
> Hey @skvijay007 did we action this?

Hi @GavinOsborn, this change is not done. I would like to use the Formula directory to avoid mixing up with higher level files

**Comment by GavinOsborn** [comment]
Cool, lets get that done then?

**Comment by skvijay007** [comment]
> Cool, lets get that done then?

Done @GavinOsborn

**Review by GavinOsborn — APPROVED** [review]

**Pull request merged**